tests/Feature/BookTest.php

```diff
@@ -79,4 +79,41 @@ public function test_getAllBooks_success(): void
             })
         ;
     }
+
+    public function test_getAllBooks_genre(): void
+    {
+        Book::factory()->count(10)->create();
+
+        $response = $this->get('/api/books?genre');
+
+        $response->assertStatus(200)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['data', 'message'])
+                    ->has('data', 10, function (AssertableJson $json) {
+                        $json->hasAll([
+                            'id',
+                            'title',
+                            'author',
+                            'image',
+                            'genre'
+                        ])->has('genre', function (AssertableJson $json) {
+                            $json->hasAll(['id', 'name']);
+                        });
+                    });
+            })
+        ;
+    }
+
+    public function test_getAllBooks_genre_error(): void
+    {
+        Book::factory()->count(10)->create();
+
+        $response = $this->get('/api/books?genre=70');
+
+        $response->assertStatus(422)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['message']);
+            })
+        ;
```

Comment:
You should use the assertInvalid method here instead - it does more checks than this + is easier https://laravel.com/docs/11.x/http-tests#validation-assert-invalid